libraries/chain/controller.cpp

```diff
@@ -3722,7 +3722,15 @@ struct controller_impl {
       block_state_ptr bsp = fork_db_fetch_bsp_on_branch_by_num(id, qc.block_num);
       if (!bsp)
          return {};
-      return bsp->aggregating_qc.vote_metrics(qc);
+
+      // Get voting metrics from QC
+      auto result =  bsp->aggregating_qc.vote_metrics(qc);
+
+      // Populate block related information
+      result.voted_block_id        = bsp->id();
+      result.voted_block_timestamp = bsp->timestamp();
```

Comment:
Done by https://github.com/AntelopeIO/spring/pull/453/commits/a1624c67b3615ece8949d310684448a266ab2375. I only added the target block number, not block ID to avoid changing `chain.missing_votes()` as it does not return `qc_vote_metrics_t` (which contains block_id).

Reply:
Maybe that should be `...  vote in block ...` instead of `on`. Also please update the `has no votes from finalizers` log on line 648.

Reply:
Done by https://github.com/AntelopeIO/spring/pull/453/commits/451e0f1abeddb6419bf339df215411b13d1afe6c